src/manage.c

```diff
@@ -2119,8 +2123,38 @@ target_osp_ssh_credential (target_t target)
           osp_credential_set_auth_data (osp_credential,
                                         "private", base64);
           g_free (base64);
+        }
+
+      if(ssh_elevate_credential)
+        {
+          const char *elevate_type;
 
+          init_credential_iterator_one (&ssh_elevate_iter,
+                                        ssh_elevate_credential);
+          if (!next (&ssh_elevate_iter))
+            {
+              g_warning ("%s: SSH Elevate Credential not found.", __func__);
+              cleanup_iterator (&ssh_elevate_iter);
+              return NULL;
```

Comment:
I think either the basic credential without the elevate data should be returned here or the SSH credential should be freed before returning NULL.